### PR TITLE
add more tests for handler and rename key range and fix parse nil hex key

### PR DESCRIPTION
**Pull request by lhy1024**

### CODE REVIEW

File: pkg/schedule/affinity/affinity.go [hunk 1]
```diff
@@ -216,7 +216,7 @@ type Manager struct {
 	affinityRegionCount int
 	groups              map[string]*GroupInfo // {group_id} -> GroupInfo
 	regions             map[uint64]regionCache
-	keyRanges           map[string][]keyRange // {group_id} -> key ranges, cached in memory to reduce labeler lock contention
+	KeyRanges           map[string][]KeyRange // {group_id} -> key ranges, cached in memory to reduce labeler lock contention
```
Comment (HunDunDM) [inline pkg/schedule/affinity/affinity.go]: `keyRanges` better?

**Review by HunDunDM — APPROVED** [review]
